runtime/compiler/env/VMJ9.cpp

```diff
@@ -764,7 +764,14 @@ TR_J9VMBase::freeSharedCache()
    {
    if (_sharedCache)        // shared classes and AOT must be enabled
       {
-      TR_ASSERT(TR::Options::sharedClassCache(), "Found shared cache with option disabled");
+      if (_compInfo && (_compInfo->getPersistentInfo()->getJITaaSMode() == SERVER_MODE))
+         {
+         TR_ASSERT(feGetEnv("TR_EnableJITaaSRemoteAOT"), "Found shared cache with remote AOT disabled.");
```

Comment:
The `freeSharedCache` method looks like it will be called every time a thread is destroyed, which can add up to a lot of calls. Unfortunately `feGetEnv` leaks memory so it's best to declare a static variable to hold the return value so that `feGetEnv` will only be called once.

```suggestion
         static bool isRemoteAOT = feGetEnv("TR_EnableJITaaSRemoteAOT") != NULL;
         TR_ASSERT(isRemoteAOT, "Found shared cache with remote AOT disabled.");
```

Can you also fix this instance while you're at it?

https://github.com/eclipse/openj9/blob/69199c55d85917c4b885ec098dfab672be263f6e/runtime/compiler/env/VMJ9.cpp#L739